tfs/src/main/java/hudson/plugins/tfs/model/AbstractHookEvent.java

```diff
@@ -81,6 +81,68 @@ static JSONObject fromResponseContributors(final List<GitStatus.ResponseContribu
         return result;
     }
 
+    GitStatus.ResponseContributor triggerJob(final GitCodePushedEventArgs gitCodePushedEventArgs, final List<Action> actions, final boolean bypassPolling, final Item project, final SCMTriggerItem scmTriggerItem) {
+        GitStatus.ResponseContributor gitResponse = null;
+        
+        if (!(project instanceof AbstractProject && ((AbstractProject) project).isDisabled())) {
+            if (project instanceof Job) {
+                // TODO: Add default parameters defined in the job
+                final Job job = (Job) project;
+                final int quietPeriod = scmTriggerItem.getQuietPeriod();                
+                
+                boolean triggered = false;
+                if (!triggered) {
+                    final TeamPluginGlobalConfig config = TeamPluginGlobalConfig.get();
+                    if (config.isEnableTeamPushTriggerForAllJobs()) {
+                        triggered = true;
+                        final SCMTrigger scmTrigger = TeamEventsEndpoint.findTrigger(job, SCMTrigger.class);
+                        if (scmTrigger != null && scmTrigger.isIgnorePostCommitHooks()) {
+                            // job has explicitly opted out of hooks
+                            triggered = false;
+                        }
+                    }
+                    if (triggered) {
+                        final TeamPushTrigger trigger = new TeamPushTrigger(job);
+                        trigger.execute(gitCodePushedEventArgs, actions, bypassPolling);
+                        if (bypassPolling) {
+                            gitResponse = new TeamEventsEndpoint.ScheduledResponseContributor(project);
+                        }
+                        else {
+                            gitResponse = new TeamEventsEndpoint.PollingScheduledResponseContributor(project);
+                        }
+                    }
+                }
+                if (!triggered) {

```

Comment:
If you inline the if (triggered) block, then this if (!triggered) becomes unnecessary since it's always false if you get here.